salt/cloud/clouds/vultrpy.py

```diff
@@ -174,7 +174,12 @@ def destroy(name):
     node = show_instance(name, call='action')
     params = {'SUBID': node['SUBID']}
     result = _query('server/destroy', method='POST', decode=False, data=urllib.urlencode(params))
-    if result['body'] == '' and result['text'] == '':
+
+    # The return of a destroy call is empty in the case of a success.
+    # Errors are only indicated via HTTP status code. Status code 200
+    # effetively therefore means "success".
+    if ('body' in result and 'text' in result and
```

Comment:
I hadn't realized that `dict.get("str")` called without specifying the default returns none (I thought it threw a `KeyError`, for some reason). Good point, I'll make the change.